guides/17_Migration_to_3_0.md

```diff
@@ -53,4 +58,68 @@ handle them please remove.
 - AudioCallEnded
 - TURNMeetingEnded
 - StateMachineTransitionFailed
-- AudioDeviceSwitched
\ No newline at end of file
+- AudioDeviceSwitched
+
+### Updating `EventController` use cases
+We have de-coupled the `EventController` from `AudioVideoController`. Check below for the new addtions.
+```typescript
+interface EventController {
+  // Adds an observer for event published to this controller.
+  addObserver(observer: EventObserver): void;
+
+  // Removes an observer for event published to this controller.
+  removeObserver(observer: EventObserver): void;
+
+  // EventReporter that the EventController uses to send events to the Amazon Chime backend.
+  readonly eventReporter?: EventReporter;
+}
+```
+
+Use case 2: Update creation of `EventController`
+```typescript
+// Before in 2.x:
+const configuration = new MeetingSessionConfiguration(…);
+const logger = new Logger(…);
+const eventReporter = new EventReporter(...)
+…
+const audioVideoController = new DefaultAudioVideoController(configuration, logger, ..., eventReporter);
+const eventController = new DefaultEventController(audioVideoController, eventReporter)
+// After in 3.x:
+const configuration = new MeetingSessionConfiguration(…);
+const logger = new Logger(…);
+const eventReporter = new EventReporter(...)
+…
+const eventController = new DefaultEventController(configuration, logger, eventReporter)
+const audioVideoController = new DefaultAudioVideoController(configuration, logger, ..., eventController);
+```
+
+### Updating `EventReporter` use cases
+
+The `DefaultMeetingSession` constructor no longer takes in a `EventReporter` and instead optionally takes in an `EventController`.
```

Comment:
You may want to call out if nothing is passed whether any default implementation be created internally.

Reply:
Yeah probably a good idea to call out, updated